### PR TITLE
    additional testcases

### PR DESCRIPTION
**Pull request by mbelur**
additional testcases
    
    Testcases:
    - simulate delayed payment and validate notification and
    clear delayed payment and validate notification cleared.
    
    - simulate expire in the past and validate notification
    and clear expire in the past and validate notification cleared.
    
    - simulate billing api access false and validate notification
    and reset the billing api access to true and validate notification cleared.
    
    - ui-brand check
    
    Added oidc.tf
    
    Checks related to pods running to the suite level.
    
    Added restartPod function to support the simulation testcases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,12 @@ To run the billing tests, you'll need to set an environment variable: CATTLE_TES
 NOTE:
 The terraform deployment scripts *automatically* create the cattle-config-e2e.yaml with the contents specific to the environment deployed.
 
-The user does not have to change this yaml file unless the user is targetting to run these tests against a different Rancher deployment other than the one deployed by terraform.
+The user does not have to change this yaml file unless the user is targetting to run these tests against a different Rancher deployment other than the one deployed by terraform. 
+
+NOTE: To run these tests successfully, the rancher-csp-billing-adapter configuration for the query_interval and reporting_interval must be set
+to a value that's reasonable for the tests to validate. Please see 
+config.yaml and update_config_csp_billing_adapter.sh to update your billing adapter deployment with the configuration especially the query_interval and the reporting_interval configuration values needed for the tests.
+An environment built with terraform does it automatically for you.
```
Comment (rtamalin) [inline README.md:43]: Minor comment nit:

```suggestion
An terraform deployment with `deploy_payg_adapter` set to true will automatically set this up for you.
```

Comment (mbelur) [reply README.md]: Updated

File: README.md [hunk 2]
```diff
@@ -98,35 +103,29 @@ The automated tests in this repo can be run on the command-line, example:
...
---- PASS: TestBillingPAYGTestSuite (847.28s)
-    --- PASS: TestBillingPAYGTestSuite/TestValidateCSPBillingAdapterPodStatus (0.77s)
-    --- PASS: TestBillingPAYGTestSuite/TestValidateCSPRancherUsageOperatorPodStatus (0.50s)
-    --- PASS: TestBillingPAYGTestSuite/TestValidateTotalCountAfterProvisioningHostedEKSCluster (844.86s)
-PASS
-ok  	csp-adapter-rancher-e2e-automation/tests	847.338s
+```
+NOTE:
+The provisioning test in the testsuite is written to clean up any resources it created in AWS.
+This cleanup in the background takes a few minutes (around 10 minutes).
```
[the remark this answers is not in the document]
Comment (mbelur) [reply README.md]: Updated

**Review by rtamalin — APPROVED** [review]
Looks good to me from a concept perspective... May want to wait for someone with better Go expertise if you want feedback on the Go code. ;-)

The one caveat is that since you are just restarting the pods, rather than updating the csp-config while the pod is down, your changes will likely get overwritten anyway query_interval seconds after the CSP Billing Adapter restarts. But at least you have a controlled synchronisation point between between you test run and the CSP Billing Adapter runtime, meaning that so long as your tests run smoothly in a timely fashion, and complete before the query_interval ends, your tests should be reliable/repeatable.

**Review by guangyee — APPROVED** [review]
lgtm

**Pull request merged**